notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/service/jpa/JpaNotificationDao.java

```diff
@@ -116,4 +119,12 @@ public void removeEntry(JpaEntry entry) {
         return rslt;
     }
 
+	@Override
+	@Transactional
+    public JpaEvent createOrUpdateEvent(JpaEvent event) {
+        Validate.notNull(event, "Argument 'event' cannot be null");
```

Comment:
JpaEvent.entry's @OneToOne relationship doesn't specify optional = false.  Now that we are creating JpaEvents that should be fixed in this pull request.  Otherwise you can create a JpaEvent that is not associated to a JpaEntry.